packages/react-server/core/renderMiddleware.js

```diff
@@ -138,7 +138,10 @@ module.exports = function(server, routes) {
 
 		var navigateDfd = Q.defer();
 
-		setTimeout(navigateDfd.reject, FAILSAFE_ROUTER_TIMEOUT);
+		const timeout = setTimeout(navigateDfd.reject, FAILSAFE_ROUTER_TIMEOUT);
```

Comment:
Shouldn't matter.  These are intra-request timers, and we're going to wait for requests to wrap up on a graceful shutdown, anyway.